synfig-studio/src/gui/app.cpp

```diff
@@ -1362,15 +1361,68 @@ App::get_default_accel_map()
 
 	return default_accel_map;
 }
+Glib::RefPtr<App> App::instance() {
+	static Glib::RefPtr<studio::App> app_reference = Glib::RefPtr<App>(new App());
+	return app_reference;
+}
 
 /* === M E T H O D S ======================================================= */
+App::App()
+	: Gtk::Application("org.synfig.SynfigStudio", Gio::APPLICATION_HANDLES_OPEN)
+{
+	add_main_option_entry(Gio::Application::OPTION_TYPE_BOOL, "console", 'c', N_("Opens a console that shows some Synfig Studio output"));
+	signal_handle_local_options().connect(sigc::mem_fun(*this, &App::on_handle_local_options));
+}
+
+int App::on_handle_local_options(const Glib::RefPtr<Glib::VariantDict> &options)
+{
+	if (!options)
+		return -1;
+#ifdef _WIN32
+	if (options->contains("console")) {
+		redirectIOToConsole();
+	} else {
+		// QuickHack: to avoid strange bug with stderr
+		freopen("NUL", "w", stdout);
+		freopen("NUL", "w", stderr);
+		freopen("NUL", "r", stdin);
+		ios::sync_with_stdio();
```

Comment:
`std::ios::sync_with_stdio`

Reply:
OK. But I don't even know why such code exists: the comment left by blackwarthog does not mention what bug it prevents: `stderr`, `stdout`, `stdin` does not even exist on Windows without the `--console` approach.

Reply:
I suggest removing this code. In any case, it is better to return the "strange" error and fix it.